src/gui/src/heatMap.h

```diff
@@ -207,7 +210,7 @@ class HeatMapDataSource
   Map map_;
 
   std::unique_ptr<HeatMapRenderer> renderer_;
-  std::unique_ptr<HeatMapSetup> setup_;
+  HeatMapSetup* setup_;
```

Comment:
Why this change?

Reply:
I got a segfault where the QT event queue still seemed to contain this Dialog after I had deleted it, this way (along with the connect statements), Qt handles the deletion in the queue in the right order.

Reply:
Where is it being deleted?

Reply:
Lines 489-490: sets up the deletion in the event queue and then when it's been destroyed (deleted), sets the pointer back to nullptr.
```
    QObject::connect(setup_, &QDialog::finished, &QObject::deleteLater);
    QObject::connect(setup_, &QObject::destroyed, [this]() { setup_ = nullptr; });
```

Reply:
This seems very odd but I don't want to spend time digging into it.